Dockerfile-local

```
@@ -17,7 +17,7 @@ ARG CP_VERSION
 
 FROM confluentinc/cp-server-connect-base:$CP_VERSION
 
-ARG CONNECTOR_VERSION=5.5.0-SNAPSHOT
+ARG CONNECTOR_VERSION=6.0.0
```

I don't think this matters too much because `Dockerfile-local` gets invoked via the start script and `CONNECTOR_VERSION` gets overridden.  However, since it gets invoked for SNAPSHOT/.x releases, then maybe this should be:

```suggestion
ARG CONNECTOR_VERSION=6.0.1-SNAPSHOT
```

---

How about we remove the defaulit entirely so we don't have to maintain it and force the builder of the image to provide it?

---

Here is a test run removing the default entirely and running scripts/start.sh

```
Building custom Docker image with Connect version 6.0.0 and connector version 6.0.0
docker build --build-arg CP_VERSION=6.0.0 --build-arg CONNECTOR_VERSION=6.0.0 -t localbuild/connect:6.0.0-6.0.0 -f /Users/rspurgeon/dev/confluentinc/cp-demo/scripts/../Dockerfile-confluenthub .
Sending build context to Docker daemon  123.1MB
Step 1/6 : ARG CP_VERSION
Step 2/6 : FROM confluentinc/cp-server-connect-base:$CP_VERSION
 ---> 3f32608ca93d
Step 3/6 : ARG CONNECTOR_VERSION=5.4.1
 ---> Using cache
 ---> 6a9186588cf2
Step 4/6 : ENV CONNECT_PLUGIN_PATH: "/usr/share/java,/connect-plugins,/usr/share/confluent-hub-components"
 ---> Using cache
 ---> d3d45535609a
Step 5/6 : RUN confluent-hub install --no-prompt confluentinc/kafka-connect-replicator:${CONNECTOR_VERSION}
 ---> Using cache
 ---> f3716a9b0071
Step 6/6 : RUN confluent-hub install --no-prompt confluentinc/kafka-connect-elasticsearch:latest
 ---> Using cache
 ---> 1c459612b0a9
Successfully built 1c459612b0a9
Successfully tagged localbuild/connect:6.0.0-6.0.0
```

---

Here is a scripts/start.sh test run w/ the default removed entirely

```
Building custom Docker image with Connect version 6.0.0 and connector version 6.0.0
docker build --build-arg CP_VERSION=6.0.0 --build-arg CONNECTOR_VERSION=6.0.0 -t localbuild/connect:6.0.0-6.0.0 -f /Users/rspurgeon/dev/confluentinc/cp-demo/scripts/../Dockerfile-confluenthub .
Sending build context to Docker daemon  123.1MB
Step 1/6 : ARG CP_VERSION
Step 2/6 : FROM confluentinc/cp-server-connect-base:$CP_VERSION
 ---> 3f32608ca93d
Step 3/6 : ARG CONNECTOR_VERSION=
 ---> Running in 59372efdb55b
Removing intermediate container 59372efdb55b
 ---> 6a8484202189
Step 4/6 : ENV CONNECT_PLUGIN_PATH: "/usr/share/java,/connect-plugins,/usr/share/confluent-hub-components"
 ---> Running in 04723e4ecc7c
Removing intermediate container 04723e4ecc7c
 ---> b6f90aba78ed
Step 5/6 : RUN confluent-hub install --no-prompt confluentinc/kafka-connect-replicator:${CONNECTOR_VERSION}
 ---> Running in ccdf2a7d5fc3
Running in a "--no-prompt" mode
Implicit acceptance of the license below:
Confluent Software Evaluation License
https://www.confluent.io/software-evaluation-license
Downloading component Confluent Kafka Replicator 6.0.0, provided by Confluent, Inc. from Confluent Hub and installing into /usr/share/confluent-hub-components
Adding installation directory to plugin path in the following files:
  /etc/kafka/connect-distributed.properties
  /etc/kafka/connect-standalone.properties
  /etc/schema-registry/connect-avro-distributed.properties
  /etc/schema-registry/connect-avro-standalone.properties

Completed
Removing intermediate container ccdf2a7d5fc3
 ---> 4a291e36332d
Step 6/6 : RUN confluent-hub install --no-prompt confluentinc/kafka-connect-elasticsearch:latest
 ---> Running in ca4be1408476
Running in a "--no-prompt" mode
Implicit acceptance of the license below:
Confluent Community License
http://www.confluent.io/confluent-community-license
Downloading component Kafka Connect Elasticsearch 10.0.0, provided by Confluent, Inc. from Confluent Hub and installing into /usr/share/confluent-hub-components
Adding installation directory to plugin path in the following files:
  /etc/kafka/connect-distributed.properties
  /etc/kafka/connect-standalone.properties
  /etc/schema-registry/connect-avro-distributed.properties
  /etc/schema-registry/connect-avro-standalone.properties

Completed
Removing intermediate container ca4be1408476
 ---> c5e36230992a
Successfully built c5e36230992a
Successfully tagged localbuild/connect:6.0.0-6.0.0
```

---

IMO having a default is better than none, even it works with no default set.  

Question is just what it should be set to.  I'm not strongly opinionated since it gets overridden anyway, but the _logical_ purpose of this file was for local SNAPSHOT.  For 6.0.0-post branch, this can point to 6.0.0; for 6.0.x branch, this can point to 6.0.0-SNAPSHOT

---

My observation is that if a user is going to override this, they do so through the script, which passes through the value.  If there isn't a clear default, then I would argue against having one at all.  What makes having a default that doesn't work across all the branches as we want better?

---

Ok, we can go either way, I defer to you.